### PR TITLE
API Gateway WebSocket chat app example

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Add an example demonstrating how to create a simple chat application using [API Gateway's WebSocket-based API](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-websocket-api.html) features, based on the [AWS Serverless Application Model](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/what-is-sam.html) [`simple-websockets-chat-app`](https://github.com/aws-samples/simple-websockets-chat-app).

Relates https://github.com/terraform-providers/terraform-provider-aws/issues/7004.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
New example: examples/api-gateway-websocket-chat-app
```

Output from acceptance testing:

```
n/a
```

Requires https://github.com/terraform-providers/terraform-provider-aws/pull/9232 to run.

### CODE REVIEW

**Comment by ewbankkit** [comment]
Updated to use released API Gateway v2 resources and latest upstream example code:

```console
$ terraform apply \
    -var="aws_access_key=yourawsaccesskey" \
    -var="aws_secret_key=yourawssecretkey" \
    -var="aws_region=us-east-1"

...

Plan: 31 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

aws_iam_role.SendMessageRole: Creating...
  arn:                   "" => "<computed>"
  assume_role_policy:    "" => "{\n  \"Version\": \"2012-10-17\",\n  \"Statement\": [{\n    \"Effect\": \"Allow\",\n    \"Principal\": {\n      \"Service\": \"lambda.amazonaws.com\"\n    },\n    \"Action\": \"sts:AssumeRole\"\n  }]\n}\n"
  create_date:           "" => "<computed>"
  force_detach_policies: "" => "false"
  max_session_duration:  "" => "3600"
  name:                  "" => "SendMessageRole"
  path:                  "" => "/"
  unique_id:             "" => "<computed>"
aws_apigatewayv2_api.SimpleChatWebSocket: Creating...
  api_endpoint:                 "" => "<computed>"
  api_key_selection_expression: "" => "$request.header.x-api-key"
  arn:                          "" => "<computed>"
  execution_arn:                "" => "<computed>"
  name:                         "" => "SimpleChatWebSocket"
  protocol_type:                "" => "WEBSOCKET"
  route_selection_expression:   "" => "$request.body.message"
aws_iam_role.OnDisconnectRole: Creating...
  arn:                   "" => "<computed>"
  assume_role_policy:    "" => "{\n  \"Version\": \"2012-10-17\",\n  \"Statement\": [{\n    \"Effect\": \"Allow\",\n    \"Principal\": {\n      \"Service\": \"lambda.amazonaws.com\"\n    },\n    \"Action\": \"sts:AssumeRole\"\n  }]\n}\n"
  create_date:           "" => "<computed>"
  force_detach_policies: "" => "false"
  max_session_duration:  "" => "3600"
  name:                  "" => "OnDisconnectRole"
  path:                  "" => "/"
  unique_id:             "" => "<computed>"
aws_iam_role.OnConnectRole: Creating...
  arn:                   "" => "<computed>"
  assume_role_policy:    "" => "{\n  \"Version\": \"2012-10-17\",\n  \"Statement\": [{\n    \"Effect\": \"Allow\",\n    \"Principal\": {\n      \"Service\": \"lambda.amazonaws.com\"\n    },\n    \"Action\": \"sts:AssumeRole\"\n  }]\n}\n"
  create_date:           "" => "<computed>"
  force_detach_policies: "" => "false"
  max_session_duration:  "" => "3600"
  name:                  "" => "OnConnectRole"
  path:                  "" => "/"
  unique_id:             "" => "<computed>"
aws_dynamodb_table.ConnectionsTable: Creating...
  arn:                                  "" => "<computed>"
  attribute.#:                          "" => "1"
  attribute.2949998522.name:            "" => "connectionId"
  attribute.2949998522.type:            "" => "S"
  billing_mode:                         "" => "PROVISIONED"
  hash_key:                             "" => "connectionId"
  name:                                 "" => "simplechat_connections"
  point_in_time_recovery.#:             "" => "<computed>"
  read_capacity:                        "" => "5"
  server_side_encryption.#:             "" => "1"
  server_side_encryption.0.enabled:     "" => "true"
  server_side_encryption.0.kms_key_arn: "" => "<computed>"
  stream_arn:                           "" => "<computed>"
  stream_label:                         "" => "<computed>"
  stream_view_type:                     "" => "<computed>"
  write_capacity:                       "" => "5"
aws_iam_role.OnDisconnectRole: Creation complete after 1s (ID: OnDisconnectRole)
aws_iam_role.OnConnectRole: Creation complete after 1s (ID: OnConnectRole)
aws_iam_role.SendMessageRole: Creation complete after 1s (ID: SendMessageRole)
aws_apigatewayv2_api.SimpleChatWebSocket: Creation complete after 1s (ID: f84w3oudm4)
aws_iam_policy.SendMessageManageConnectionsPolicy: Creating...
  arn:    "" => "<computed>"
  name:   "" => "SendMessageManageConnectionsPolicy"
  path:   "" => "/"
  policy: "" => "{\n  \"Version\": \"2012-10-17\",\n  \"Statement\": [{\n    \"Effect\": \"Allow\",\n    \"Action\": \"execute-api:ManageConnections\",\n    \"Resource\": \"arn:aws:execute-api:us-east-1:123456789012:f84w3oudm4/*\"\n  }]\n}\n"
aws_iam_policy.SendMessageManageConnectionsPolicy: Creation complete after 0s (ID: arn:aws:iam::123456789012:policy/SendMessageManageConnectionsPolicy)
aws_iam_role_policy_attachment.SendMessageRoleSendMessageManageConnectionsPolicyAttachment: Creating...
  policy_arn: "" => "arn:aws:iam::123456789012:policy/SendMessageManageConnectionsPolicy"
  role:       "" => "SendMessageRole"
aws_iam_role_policy_attachment.SendMessageRoleSendMessageManageConnectionsPolicyAttachment: Creation complete after 0s (ID: SendMessageRole-20200419203638264200000001)
aws_dynamodb_table.ConnectionsTable: Creation complete after 7s (ID: simplechat_connections)
aws_iam_policy.DynamoDBCrudPolicy: Creating...
  arn:    "" => "<computed>"
  name:   "" => "DynamoDBCrudPolicy"
  path:   "" => "/"
  policy: "" => "{\n  \"Version\": \"2012-10-17\",\n  \"Statement\": [{\n    \"Effect\": \"Allow\",\n    \"Action\": [\n      \"dynamodb:GetItem\",\n      \"dynamodb:DeleteItem\",\n      \"dynamodb:PutItem\",\n      \"dynamodb:Scan\",\n      \"dynamodb:Query\",\n      \"dynamodb:UpdateItem\",\n      \"dynamodb:BatchWriteItem\",\n      \"dynamodb:BatchGetItem\",\n      \"dynamodb:DescribeTable\"\n    ],\n    \"Resource\": [\n      \"arn:aws:dynamodb:us-east-1:123456789012:table/simplechat_connections\",\n      \"arn:aws:dynamodb:us-east-1:123456789012:table/simplechat_connections/index/*\"\n    ]\n  }]\n}\n"
aws_lambda_function.SendMessageFunction: Creating...
  arn:                                "" => "<computed>"
  environment.#:                      "" => "1"
  environment.0.variables.%:          "" => "1"
  environment.0.variables.TABLE_NAME: "" => "simplechat_connections"
  filename:                           "" => "/home/kit/wrk/src/github.com/terraform-providers/terraform-provider-aws/examples/api-gateway-websocket-chat-app/sendmessage/app.zip"
  function_name:                      "" => "SendMessageFunction"
  handler:                            "" => "app.handler"
  invoke_arn:                         "" => "<computed>"
  last_modified:                      "" => "<computed>"
  memory_size:                        "" => "256"
  publish:                            "" => "false"
  qualified_arn:                      "" => "<computed>"
  reserved_concurrent_executions:     "" => "-1"
  role:                               "" => "arn:aws:iam::123456789012:role/SendMessageRole"
  runtime:                            "" => "nodejs12.x"
  source_code_hash:                   "" => "<computed>"
  source_code_size:                   "" => "<computed>"
  timeout:                            "" => "3"
  tracing_config.#:                   "" => "<computed>"
  version:                            "" => "<computed>"
aws_lambda_function.OnConnectFunction: Creating...
  arn:                                "" => "<computed>"
  environment.#:                      "" => "1"
  environment.0.variables.%:          "" => "1"
  environment.0.variables.TABLE_NAME: "" => "simplechat_connections"
  filename:                           "" => "/home/kit/wrk/src/github.com/terraform-providers/terraform-provider-aws/examples/api-gateway-websocket-chat-app/onconnect/app.zip"
  function_name:                      "" => "OnConnectFunction"
  handler:                            "" => "app.handler"
  invoke_arn:                         "" => "<computed>"
  last_modified:                      "" => "<computed>"
  memory_size:                        "" => "256"
  publish:                            "" => "false"
  qualified_arn:                      "" => "<computed>"
  reserved_concurrent_executions:     "" => "-1"
  role:                               "" => "arn:aws:iam::123456789012:role/OnConnectRole"
  runtime:                            "" => "nodejs12.x"
  source_code_hash:                   "" => "<computed>"
  source_code_size:                   "" => "<computed>"
  timeout:                            "" => "3"
  tracing_config.#:                   "" => "<computed>"
  version:                            "" => "<computed>"
aws_lambda_function.OnDisconnectFunction: Creating...
  arn:                                "" => "<computed>"
  environment.#:                      "" => "1"
  environment.0.variables.%:          "" => "1"
  environment.0.variables.TABLE_NAME: "" => "simplechat_connections"
  filename:                           "" => "/home/kit/wrk/src/github.com/terraform-providers/terraform-provider-aws/examples/api-gateway-websocket-chat-app/ondisconnect/app.zip"
  function_name:                      "" => "OnDisconnectFunction"
  handler:                            "" => "app.handler"
  invoke_arn:                         "" => "<computed>"
  last_modified:                      "" => "<computed>"
  memory_size:                        "" => "256"
  publish:                            "" => "false"
  qualified_arn:                      "" => "<computed>"
  reserved_concurrent_executions:     "" => "-1"
  role:                               "" => "arn:aws:iam::123456789012:role/OnDisconnectRole"
  runtime:                            "" => "nodejs12.x"
  source_code_hash:                   "" => "<computed>"
  source_code_size:                   "" => "<computed>"
  timeout:                            "" => "3"
  tracing_config.#:                   "" => "<computed>"
  version:                            "" => "<computed>"
aws_iam_policy.DynamoDBCrudPolicy: Creation complete after 0s (ID: arn:aws:iam::123456789012:policy/DynamoDBCrudPolicy)
aws_iam_role_policy_attachment.OnConnectRoleDynamoDBCrudPolicyAttachment: Creating...
  policy_arn: "" => "arn:aws:iam::123456789012:policy/DynamoDBCrudPolicy"
  role:       "" => "OnConnectRole"
aws_iam_role_policy_attachment.OnDisconnectRoleDynamoDBCrudPolicyAttachment: Creating...
  policy_arn: "" => "arn:aws:iam::123456789012:policy/DynamoDBCrudPolicy"
  role:       "" => "OnDisconnectRole"
aws_iam_role_policy_attachment.SendMessageRoleDynamoDBCrudPolicyAttachment: Creating...
  policy_arn: "" => "arn:aws:iam::123456789012:policy/DynamoDBCrudPolicy"
  role:       "" => "SendMessageRole"
aws_iam_role_policy_attachment.SendMessageRoleDynamoDBCrudPolicyAttachment: Creation complete after 0s (ID: SendMessageRole-20200419203644901400000002)
aws_iam_role_policy_attachment.OnDisconnectRoleDynamoDBCrudPolicyAttachment: Creation complete after 0s (ID: OnDisconnectRole-20200419203644934300000003)
aws_iam_role_policy_attachment.OnConnectRoleDynamoDBCrudPolicyAttachment: Creation complete after 0s (ID: OnConnectRole-20200419203644934900000004)
aws_lambda_function.SendMessageFunction: Creation complete after 7s (ID: SendMessageFunction)
aws_apigatewayv2_integration.SendInteg: Creating...
  api_id:                                    "" => "f84w3oudm4"
  connection_type:                           "" => "INTERNET"
  description:                               "" => "Send Integration"
  integration_method:                        "" => "POST"
  integration_response_selection_expression: "" => "<computed>"
  integration_type:                          "" => "AWS_PROXY"
  integration_uri:                           "" => "arn:aws:apigateway:us-east-1:lambda:path/2015-03-31/functions/arn:aws:lambda:us-east-1:123456789012:function:SendMessageFunction/invocations"
  passthrough_behavior:                      "" => "WHEN_NO_MATCH"
  payload_format_version:                    "" => "1.0"
  timeout_milliseconds:                      "" => "29000"
aws_lambda_permission.SendMessagePermission: Creating...
  action:        "" => "lambda:InvokeFunction"
  function_name: "" => "SendMessageFunction"
  principal:     "" => "apigateway.amazonaws.com"
  statement_id:  "" => "<computed>"
aws_iam_policy.SendMessageCloudWatchLogsPolicy: Creating...
  arn:    "" => "<computed>"
  name:   "" => "SendMessageCloudWatchLogsPolicy"
  path:   "" => "/"
  policy: "" => "{\n  \"Version\": \"2012-10-17\",\n  \"Statement\": [\n    {\n      \"Effect\": \"Allow\",\n      \"Action\": \"logs:CreateLogGroup\",\n      \"Resource\": \"arn:aws:logs:us-east-1:123456789012:*\"\n    },\n    {\n      \"Effect\": \"Allow\",\n      \"Action\": [\n        \"logs:CreateLogStream\",\n        \"logs:PutLogEvents\"\n      ],\n      \"Resource\": [\n        \"arn:aws:logs:us-east-1:123456789012:log-group:/aws/lambda/SendMessageFunction:*\"\n      ]\n    }\n  ]\n}\n"
aws_lambda_permission.SendMessagePermission: Creation complete after 0s (ID: terraform-20200419203651938000000005)
aws_iam_policy.SendMessageCloudWatchLogsPolicy: Creation complete after 0s (ID: arn:aws:iam::123456789012:policy/SendMessageCloudWatchLogsPolicy)
aws_iam_role_policy_attachment.SendMessageRoleSendMessageCloudWatchLogsPolicyAttachment: Creating...
  policy_arn: "" => "arn:aws:iam::123456789012:policy/SendMessageCloudWatchLogsPolicy"
  role:       "" => "SendMessageRole"
aws_apigatewayv2_integration.SendInteg: Creation complete after 0s (ID: tp8ner7)
aws_apigatewayv2_route.SendRoute: Creating...
  api_id:             "" => "f84w3oudm4"
  api_key_required:   "" => "false"
  authorization_type: "" => "NONE"
  operation_name:     "" => "SendRoute"
  route_key:          "" => "sendmessage"
  target:             "" => "integrations/tp8ner7"
aws_iam_role_policy_attachment.SendMessageRoleSendMessageCloudWatchLogsPolicyAttachment: Creation complete after 0s (ID: SendMessageRole-20200419203652264100000006)
aws_apigatewayv2_route.SendRoute: Creation complete after 1s (ID: fdt4628)
aws_lambda_function.OnConnectFunction: Still creating... (10s elapsed)
aws_lambda_function.OnDisconnectFunction: Still creating... (10s elapsed)
aws_lambda_function.OnConnectFunction: Creation complete after 12s (ID: OnConnectFunction)
aws_lambda_permission.OnConnectPermission: Creating...
  action:        "" => "lambda:InvokeFunction"
  function_name: "" => "OnConnectFunction"
  principal:     "" => "apigateway.amazonaws.com"
  statement_id:  "" => "<computed>"
aws_iam_policy.OnConnectCloudWatchLogsPolicy: Creating...
  arn:    "" => "<computed>"
  name:   "" => "OnConnectCloudWatchLogsPolicy"
  path:   "" => "/"
  policy: "" => "{\n  \"Version\": \"2012-10-17\",\n  \"Statement\": [\n    {\n      \"Effect\": \"Allow\",\n      \"Action\": \"logs:CreateLogGroup\",\n      \"Resource\": \"arn:aws:logs:us-east-1:123456789012:*\"\n    },\n    {\n      \"Effect\": \"Allow\",\n      \"Action\": [\n        \"logs:CreateLogStream\",\n        \"logs:PutLogEvents\"\n      ],\n      \"Resource\": [\n        \"arn:aws:logs:us-east-1:123456789012:log-group:/aws/lambda/OnConnectFunction:*\"\n      ]\n    }\n  ]\n}\n"
aws_apigatewayv2_integration.ConnectIntegrat: Creating...
  api_id:                                    "" => "f84w3oudm4"
  connection_type:                           "" => "INTERNET"
  description:                               "" => "Connect Integration"
  integration_method:                        "" => "POST"
  integration_response_selection_expression: "" => "<computed>"
  integration_type:                          "" => "AWS_PROXY"
  integration_uri:                           "" => "arn:aws:apigateway:us-east-1:lambda:path/2015-03-31/functions/arn:aws:lambda:us-east-1:123456789012:function:OnConnectFunction/invocations"
  passthrough_behavior:                      "" => "WHEN_NO_MATCH"
  payload_format_version:                    "" => "1.0"
  timeout_milliseconds:                      "" => "29000"
aws_lambda_permission.OnConnectPermission: Creation complete after 0s (ID: terraform-20200419203657503800000007)
aws_iam_policy.OnConnectCloudWatchLogsPolicy: Creation complete after 0s (ID: arn:aws:iam::123456789012:policy/OnConnectCloudWatchLogsPolicy)
aws_iam_role_policy_attachment.OnConnectRoleOnConnectCloudWatchLogsPolicyAttachment: Creating...
  policy_arn: "" => "arn:aws:iam::123456789012:policy/OnConnectCloudWatchLogsPolicy"
  role:       "" => "OnConnectRole"
aws_apigatewayv2_integration.ConnectIntegrat: Creation complete after 0s (ID: cjx5ddk)
aws_apigatewayv2_route.ConnectRoute: Creating...
  api_id:             "" => "f84w3oudm4"
  api_key_required:   "" => "false"
  authorization_type: "" => "NONE"
  operation_name:     "" => "ConnectRoute"
  route_key:          "" => "$connect"
  target:             "" => "integrations/cjx5ddk"
aws_iam_role_policy_attachment.OnConnectRoleOnConnectCloudWatchLogsPolicyAttachment: Creation complete after 0s (ID: OnConnectRole-20200419203657844600000008)
aws_apigatewayv2_route.ConnectRoute: Creation complete after 0s (ID: 5bmd4e6)
aws_lambda_function.OnDisconnectFunction: Creation complete after 18s (ID: OnDisconnectFunction)
aws_lambda_permission.OnDisconnectPermission: Creating...
  action:        "" => "lambda:InvokeFunction"
  function_name: "" => "OnDisconnectFunction"
  principal:     "" => "apigateway.amazonaws.com"
  statement_id:  "" => "<computed>"
aws_iam_policy.OnDisconnectCloudWatchLogsPolicy: Creating...
  arn:    "" => "<computed>"
  name:   "" => "OnDisconnectCloudWatchLogsPolicy"
  path:   "" => "/"
  policy: "" => "{\n  \"Version\": \"2012-10-17\",\n  \"Statement\": [\n    {\n      \"Effect\": \"Allow\",\n      \"Action\": \"logs:CreateLogGroup\",\n      \"Resource\": \"arn:aws:logs:us-east-1:123456789012:*\"\n    },\n    {\n      \"Effect\": \"Allow\",\n      \"Action\": [\n        \"logs:CreateLogStream\",\n        \"logs:PutLogEvents\"\n      ],\n      \"Resource\": [\n        \"arn:aws:logs:us-east-1:123456789012:log-group:/aws/lambda/OnDisconnectFunction:*\"\n      ]\n    }\n  ]\n}\n"
aws_apigatewayv2_integration.DisconnectInteg: Creating...
  api_id:                                    "" => "f84w3oudm4"
  connection_type:                           "" => "INTERNET"
  description:                               "" => "Disconnect Integration"
  integration_method:                        "" => "POST"
  integration_response_selection_expression: "" => "<computed>"
  integration_type:                          "" => "AWS_PROXY"
  integration_uri:                           "" => "arn:aws:apigateway:us-east-1:lambda:path/2015-03-31/functions/arn:aws:lambda:us-east-1:123456789012:function:OnDisconnectFunction/invocations"
  passthrough_behavior:                      "" => "WHEN_NO_MATCH"
  payload_format_version:                    "" => "1.0"
  timeout_milliseconds:                      "" => "29000"
aws_lambda_permission.OnDisconnectPermission: Creation complete after 0s (ID: terraform-20200419203703094400000009)
aws_iam_policy.OnDisconnectCloudWatchLogsPolicy: Creation complete after 0s (ID: arn:aws:iam::123456789012:policy/OnDisconnectCloudWatchLogsPolicy)
aws_iam_role_policy_attachment.OnDisconnectRoleOnDisconnectCloudWatchLogsPolicyAttachment: Creating...
  policy_arn: "" => "arn:aws:iam::123456789012:policy/OnDisconnectCloudWatchLogsPolicy"
  role:       "" => "OnDisconnectRole"
aws_apigatewayv2_integration.DisconnectInteg: Creation complete after 0s (ID: f1b9rc5)
aws_apigatewayv2_route.DisconnectRoute: Creating...
  api_id:             "" => "f84w3oudm4"
  api_key_required:   "" => "false"
  authorization_type: "" => "NONE"
  operation_name:     "" => "DisconnectRoute"
  route_key:          "" => "$disconnect"
  target:             "" => "integrations/f1b9rc5"
aws_iam_role_policy_attachment.OnDisconnectRoleOnDisconnectCloudWatchLogsPolicyAttachment: Creation complete after 1s (ID: OnDisconnectRole-2020041920370347930000000a)
aws_apigatewayv2_route.DisconnectRoute: Creation complete after 1s (ID: 7oylwc5)
aws_apigatewayv2_deployment.Deployment: Creating...
  api_id:        "" => "f84w3oudm4"
  auto_deployed: "" => "<computed>"
aws_apigatewayv2_deployment.Deployment: Creation complete after 1s (ID: i0nnt2)
aws_apigatewayv2_stage.Stage: Creating...
  api_id:        "" => "f84w3oudm4"
  arn:           "" => "<computed>"
  auto_deploy:   "" => "false"
  deployment_id: "" => "i0nnt2"
  description:   "" => "Prod Stage"
  execution_arn: "" => "<computed>"
  invoke_url:    "" => "<computed>"
  name:          "" => "Prod"
aws_apigatewayv2_stage.Stage: Creation complete after 0s (ID: Prod)

Apply complete! Resources: 31 added, 0 changed, 0 destroyed.

Outputs:

ConnectionsTableArn = arn:aws:dynamodb:us-east-1:123456789012:table/simplechat_connections
OnConnectFunctionArn = arn:aws:lambda:us-east-1:123456789012:function:OnConnectFunction
OnDisconnectFunctionArn = arn:aws:lambda:us-east-1:123456789012:function:OnDisconnectFunction
SendMessageFunctionArn = arn:aws:lambda:us-east-1:123456789012:function:SendMessageFunction
WebSocketURI = wss://f84w3oudm4.execute-api.us-east-1.amazonaws.com/Prod

$ wscat -c wss://f84w3oudm4.execute-api.us-east-1.amazonaws.com/Prod
Connected (press CTRL+C to quit)
> {"message":"sendmessage", "data":"hello world"}
< hello world
```

**Comment by ewbankkit** [comment]
Revisit considering https://github.com/terraform-providers/terraform-provider-aws/pull/13196.

File: examples/api-gateway-websocket-chat-app/variables.tf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+variable "aws_access_key" {
+  description = ""
+}
+
+variable "aws_secret_key" {
+  description = ""
+}
+
+variable "aws_region" {
+  description = ""
```
Comment (gdavison) [inline examples/api-gateway-websocket-chat-app/variables.tf:10]: Can you add a default value for `aws_region`? This should be as out-of-the-box as possible

File: examples/api-gateway-websocket-chat-app/main.tf [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+# Terraform configuration for simple-websockets-chat-app that has the DynamoDB table and
+# Lambda functions needed to demonstrate the Websocket protocol on API Gateway.
+#
+
+#
+# Providers.
+#
+
+provider "aws" {
+  access_key = "${var.aws_access_key}"
```
[the remark this answers is not in the document]
Comment (ewbankkit) [reply examples/api-gateway-websocket-chat-app/main.tf:11]: Will do. I'll remove the `aws_access_key` and `aws_secret_key` variables as well and update the README.

File: examples/api-gateway-websocket-chat-app/variables.tf (hunk 1 above)
Comment (ewbankkit) [reply examples/api-gateway-websocket-chat-app/variables.tf:10]: I'll default it to `us-west-2`.

**Comment by ewbankkit** [comment]
Changes as suggested but I am now getting a error

```console
$ wscat -c wss://5jbznf5gfj.execute-api.us-east-1.amazonaws.com/Prod
Connected (press CTRL+C to quit)
> hello
< {"message": "Forbidden", "connectionId":"NQdxWff6IAMAbyw=", "requestId":"NQdx2FWcoAMFZww="}
```

I'll investigate.

**Comment by ewbankkit** [comment]
@gdavison Tweaks done.

**Pull request merged**